plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/CodeWhispererFeatureConfigService.kt

```diff
@@ -82,6 +83,17 @@ class CodeWhispererFeatureConfigService {
 
     fun getChatWSContext(): Boolean = getFeatureValueForKey(CHAT_WS_CONTEXT).stringValue() == "TREATMENT"
 
+    // convert into mynahUI parsable string
+    // format: '[["key1", {"name":"Feature1","variation":"A","value":true}]]'
+    fun getFeatureConfigJsonString(): String {
+        val jsonString = featureConfigs.entries.map { (key, value) ->
+            "[\"$key\",${Gson().toJson(value)}]"
```

Comment:
If I use objectMapper here then this function would be blocked on IO forever and webview will never load.

Reply:
and Gson formatted string happens to match what MynahUI requires.

Reply:
This data in VSC is a JSON.stringfied(JSON.stringfield(Object)). See https://github.com/aws/aws-toolkit-vscode/blob/bb11bd9a4b498b2129085f43217b6b712dbf8c44/packages/amazonq/src/lsp/chat/webviewProvider.ts#L85.

Reply:
If I implement it in this way
```
val jsonString = featureConfigs.entries.map { (key, value) ->
            "[\"$key\",${jacksonObjectMapper().writeValueAsString(value)}]"
        }
```

then it would stuck at `public native void park(boolean isAbsolute, long time);` of Unsafe.java. It may have sometime to do with featureConfigs list being not thread safe but I don't find a easy fix.

Reply:
I wonder why the key value pairs are not modelled as json in mynah-ui